login_flows/santander.py

```diff
@@ -0,0 +1,123 @@
+available_authenticators = MitIDClient.identify_as_user_and_get_available_authenticators(user_id)
+
+print(f"Available authenticator: {available_authenticators}")
+
+if args.method == "TOKEN":
+    token_digits = input("Please input the 6 digits from your code token\n").strip()
+    MitIDClient.authenticate_with_token(token_digits)
+
+    password = input("Please input your password\n").strip()
+    MitIDClient.authenticate_with_password(password)
+elif args.method == "APP":
+    MitIDClient.authenticate_with_app()
+else:
+    raise Exception(f"Unknown authenticator method {args.method}")
+
+authorization_code = MitIDClient.finalize_authentication_and_get_authorization_code()
+
+print(f"Your MitID authorization code was ({authorization_code})")
+
+# Second part of Santander procedure
+payload = f'''-----------------------------{form_digits}\r\nContent-Disposition: form-data; name="authCode"\r\n\r\n{authorization_code}\r\n-----------------------------{form_digits}--\r\n'''
+
+headers = {'Content-Type': f'multipart/form-data; boundary=---------------------------{form_digits}'}
+request = session.post(soup.div.next['data-base-url']+soup.div.next['data-auth-code-path'], data=payload, headers=headers)
+request = session.get(soup.div.next['data-base-url']+soup.div.next['data-finalize-auth-path'])
+soup = BeautifulSoup(request.text, 'lxml')
+inputs = soup.find_all('input')
+json_payload = {}
+for input in inputs:
+    json_payload[input['name']] = input['value']
+
+payload = f"code={json_payload['code']}&session_state={json_payload['session_state']}&state={json_payload['state']}"
+headers = {'Content-Type': 'application/x-www-form-urlencoded'}
+request = session.post('https://oauth.scb.nu/oauth/v2/authentication/scb-obas-denmark/callback', data=payload, headers=headers)
+soup = BeautifulSoup(request.text, 'lxml')
+inputs = soup.find_all('input')
+json_payload = {}
+for input in inputs:
+    try:
+        json_payload[input['name']] = input['value']
+    except:
+        pass
+
+payload = f"token={json_payload['token']}&state={json_payload['state']}"
+request = session.post('https://oauth.scb.nu' + soup.form['action'], data=payload, headers=headers)
+code = request.url[request.url.index('code=')+5:request.url.index('&',request.url.index('code='))]
+
+payload = f"client_id=SCBDK.Netbank.Client&redirect_uri=https%3A%2F%2Fnetbank.santanderconsumer.dk%2Fdk%2Fauth%2Fcallback&grant_type=authorization_code&code={code}&code_verifier={code_verifier}"
+
+headers = {'Content-Type': 'application/x-www-form-urlencoded;charset=UTF-8'}
+request = session.post('https://oauth.scb.nu/oauth/v2/token', data=payload, headers=headers)
+request_json = request.json()
+
+access_token = request_json['access_token']
+
+headers = {
+    'User-Agent': 'Mozilla/5.0 (Windows NT 10.0; rv:126.0) Gecko/20100101 Firefox/126.0',
+    'Accept': 'application/json',
+    'Ocp-Apim-Subscription-Key': 'e11f034d95144b7a9f1427932fccf853',
```

Comment:
Can you confirm that this is a general value and not exclusive to your account?

Reply:
Yes, you can try requesting https://netbank.santanderconsumer.dk/dk/login in your browser. In developer tools, you will see a request for https://netbank.santanderconsumer.dk/702ea43f0e6ce1c87b1f.conf

The response contains the API key.